File: DocumentManagementSystem.Models/DocumentHistory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DocumentManagementSystem.Models.Common;

namespace DocumentManagementSystem.Models
{
    public class DocumentHistory
    {
        public int Id { get; set; }
        public int DocumentId { get; set; }
        public Helper.HistoryAction ActionEvent { get; set; }
        public int UserId { get; set; }
        public DateTime Date { get; set; }

        public virtual Account Account { get; set; }
        public virtual Document Document { get; set; }
    }
}

File: DocumentManagementSystem.WebApp/Content/models/documentHistory.ts
module rootApp.model {
    import User = rootApp.model.User;

    export enum HistoryAction {
        Upload = 1,
        Edit = 2,
        Delete = 3
    }

    export class DocumentHistory {
        id?: number;
        documentId?: number;
        actionEvent?: HistoryAction;
        account?: User;
        date?: Date;


        constructor(id: number, documentId: number, actionEvent: HistoryAction, account: User, date: Date) {
            this.id = id;
            this.documentId = documentId;
            this.actionEvent = actionEvent;
            this.account = account;
            this.date = date;
        }
    }
}